### PR TITLE
buffer/erase: add more corner case TCs

### PR DESCRIPTION
**Pull request by pangwon**
Buffer에서 Erase관련 TC들 추가 및 pass 하였습니다.

**EraseMergeEraseTest**
범위가 겹치는 앞선 erase가 잘 merge 되는지 test
**EraseMergeOutOfRangeTest**
10을 초과하는 경우 erase끼리 merge되지 않는지 test
**EraseMergeSingleWriteTest**
앞선 write를 erase가 잘 merge하는지 test
**EraseMergeMultipleWriteTest**
여러개의 앞선 write를 잘 merge하는지 test
**EraseNotMergeEraseBeforeWriteTest**
E - W - E 순서이고 앞서 동일 범위에 Write after erase일 경우 merge하지 않는지 test

### CODE REVIEW

File: SSD/SSD-DirtySweeper/ssd.cpp [hunk 1]
```diff
@@ -414,7 +414,7 @@ class BufferedSSD : public SSD {
 	}
 
 	std::pair<int, int> checkOverlap(int a, int b, int c, int d) {
-		if (b >= c && a <= d) {
+		if (b + 1 >= c && a <= d + 1) {
```
Comment (baencho) [inline SSD/SSD-DirtySweeper/ssd.cpp:417]: parameter를 a, b, c, d로 받고 있는데 각 parameter 의 의미가 잘 드러나는 이름이면 더 이해하기 쉬울 것 같습니다!

File: SSD/SSD-DirtySweeper/main.cpp [hunk 2]
```diff
@@ -548,6 +548,99 @@ TEST_F(BufSSDTest, Erase_Exception) {
     EXPECT_TRUE(checkOutputFile("ERROR"));
 }
 
+TEST_F(BufSSDTest, EraseMergeEraseTest) {
+    lba = 30;
```
Comment (baencho) [inline SSD/SSD-DirtySweeper/main.cpp:552]: TC 가 늘어나서 더 꼼꼼하게 예외상황을 테스트해볼 수 있을 것 같네요!

File: SSD/SSD-DirtySweeper/main.cpp [hunk 3]
```diff
@@ -548,6 +548,99 @@ TEST_F(BufSSDTest, Erase_Exception) {
     EXPECT_TRUE(checkOutputFile("ERROR"));
 }
 
+TEST_F(BufSSDTest, EraseMergeEraseTest) {
```
Comment (yalex-kim) [inline SSD/SSD-DirtySweeper/main.cpp:551]: 여러가지 Test Case 에 대해서 추가되면서 버그를 발견하신 것 같네요. 
TC 만드는 게 참 귀찮고 어려운 일인데 고민해 주셔서 감사합니다. :)

**Review by baencho — APPROVED** [review]

**Review by yalex-kim — APPROVED** [review]

File: SSD/SSD-DirtySweeper/ssd.cpp (hunk 1 above)
Comment (pangwon) [reply SSD/SSD-DirtySweeper/ssd.cpp:417]: 넵 피드백 감사합니다 리팩토링때 고려해보겠습니다!
